### DISCUSSION TITLE
Lost installer code recovery script

### DISCUSSION THREAD

**Discussion by iggut** [Dilbert66/esphome-dsckeybus#50] [Q&A] [answered]
I am trying to bruteforce the code as to not reset the current programming (previous house owner did not document anything). A person on reddit made a script for Envisalink to do this exact task!
Any chance someone that knows code could make it work for Dilbert66's dsc keybus integration if it is possible?
[bfINSTpassword.txt](https://github.com/Dilbert66/esphome-dsckeybus/files/8985253/bfINSTpassword.txt)

**Comment by Dilbert66** [comment, marked as answer]
Try this to unlock a DSC system.  This is the source library my code is based on

https://github.com/taligentx/dscKeybusInterface/tree/master/examples/esp8266/Unlocker

**Comment by iggut** [comment]
Got Arduino installed and compiled/uploaded the unlocker sketch you linked. I am very new at this. By looking at the code the LED will blink when it finds the code.
Is there a way to connect to the nodemcu and see this in a terminal? I no longer see the device in Homeassistant/ ESPhome page since it was flashed by arduino

**Reply by iggut** [reply]
Would have been cool to see it happen 👯 lol
....took about 10 minutes of me wondering if it is doing something to go look at the nodemcu in the basement. Code was found and flashing the LED away :D tytyty

**Reply by Dilbert66** [reply]
You need a serial monitor connected to the usb port of the esp.  Read up on Arduino debuggging.
